app/scripts/lib/snap-keyring/metrics.ts

```diff
@@ -25,28 +28,16 @@ export async function getSnapAndHardwareInfoForMetrics(
     return {};
   }
 
-  const selectedAddress = getSelectedAddress();
+  const account = messenger.call('AccountsController:getSelectedAccount');
+  const selectedAddress = account.address;
+  const { keyring } = account.metadata;
 
-  const keyring: any = await getKeyringForAccount(selectedAddress);
-
-  const account = await getAccountFromAddress(selectedAddress);
-
-  const snap: Snap = (await messenger.call(
-    'SnapController:get',
-    account?.metadata.snap.id,
-  )) as Snap;
-
-  async function getAccountFromAddress(address: string) {
-    const lowercaseAddress = address.toLowerCase();
-
-    if (keyring.listAccounts) {
-      const accounts = await keyring.listAccounts();
-      return accounts.find(
-        (_account: any) => _account.address.toLowerCase() === lowercaseAddress,
-      );
-    }
-
-    return undefined;
+  let snap;
+  if (account.metadata?.snap?.id) {
```

Comment:
Nit: Is the first `?.` necessary here? It seems like an internal account always has `metadata`, is that true?

``` suggestion
  if (account.metadata.snap?.id) {

Reply:
Not necessary, metadata is not an optional property. Solved in [e07ed43](https://github.com/MetaMask/metamask-extension/pull/21437/commits/e07ed43b782c6563a08731bbe41e08dcbdbadcaf)